cpp/tests/lead_lag/lead_lag_test.cpp

```diff
@@ -0,0 +1,315 @@
+};
+
+using TypesForTest = cudf::test::Concat<cudf::test::IntegralTypes,
+                                        cudf::test::FloatingPointTypes,
+                                        cudf::test::DurationTypes,
+                                        cudf::test::TimestampTypes>;
+
+TYPED_TEST_CASE(TypedLeadLagWindowTest, TypesForTest);
+
+TYPED_TEST(TypedLeadLagWindowTest, LeadLagBasics)
+{
+  using T = int32_t;
+
+  auto const input_col =
+    fixed_width_column_wrapper<T>{0, 1, 2, 3, 4, 5, 0, 10, 20, 30, 40, 50}.release();
+  auto const input_size   = input_col->size();
+  auto const grouping_key = fixed_width_column_wrapper<int32_t>{0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1};
+  auto const grouping_keys = cudf::table_view{std::vector<cudf::column_view>{grouping_key}};
+
+  auto lead_3_output_col = cudf::grouped_rolling_window(
+    grouping_keys, input_col->view(), 4, 3, 1, cudf::make_lead_aggregation(3));
+
+  expect_columns_equivalent(
+    *lead_3_output_col,
+    fixed_width_column_wrapper<T>{{3, 4, 5, -1, -1, -1, 30, 40, 50, -1, -1, -1},
+                                  {1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0}}
+      .release()
+      ->view());
+
+  auto lag_2_output_col = cudf::grouped_rolling_window(
+    grouping_keys, input_col->view(), 3, 2, 1, cudf::make_lag_aggregation(2));
+
+  expect_columns_equivalent(
+    *lag_2_output_col,
+    fixed_width_column_wrapper<T>{{-1, -1, 0, 1, 2, 3, -1, -1, 0, 10, 20, 30},
+                                  {0, 0, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1}}
+      .release()
+      ->view());
+}
+
+TYPED_TEST(TypedLeadLagWindowTest, LeadLagWithNulls)
+{
+  using T = TypeParam;
+
+  auto const input_col = fixed_width_column_wrapper<T>{{0, 1, 2, 3, 4, 5, 0, 10, 20, 30, 40, 50},
+                                                       {1, 1, 0, 1, 1, 1, 1, 1, 0, 1, 1, 1}}
+                           .release();
+  auto const input_size   = input_col->size();
+  auto const grouping_key = fixed_width_column_wrapper<int32_t>{0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1};
+  auto const grouping_keys = cudf::table_view{std::vector<cudf::column_view>{grouping_key}};
+
+  auto lead_3_output_col = cudf::grouped_rolling_window(
+    grouping_keys, input_col->view(), 4, 3, 1, cudf::make_lead_aggregation(3));
+
+  expect_columns_equivalent(
+    *lead_3_output_col,
+    fixed_width_column_wrapper<T>{{3, 4, 5, -1, -1, -1, 30, 40, 50, -1, -1, -1},
+                                  {1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0}}
+      .release()
+      ->view());
+
+  auto const lag_2_output_col = cudf::grouped_rolling_window(
+    grouping_keys, input_col->view(), 3, 2, 1, cudf::make_lag_aggregation(2));
+
+  expect_columns_equivalent(
+    *lag_2_output_col,
+    fixed_width_column_wrapper<T>{{-1, -1, 0, 1, -1, 3, -1, -1, 0, 10, -1, 30},
+                                  {0, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 1}}
+      .release()
+      ->view());
+}
+
+TYPED_TEST(TypedLeadLagWindowTest, TestLeadLagWithDefaults)
```

Comment:
I don't see any test case for defaults having null values and non values.

Reply:
I have added such a test now.